src/Events/Integrations/Modules/Ticket_Emails/Email/RSVP.php

```diff
@@ -133,47 +133,75 @@ public function include_attachments( $attachments, $email_id, $email_class ) {
 	 * @return void
 	 */
 	public function include_event_links( $parent_template ) {
-		$email_class = tribe( RSVP_Email::class );
+		if ( ! $this->should_show_links( $parent_template, $email_class, $args ) ) {
+			return;
+		}
 
-		// Bail early if the email class is not enabled.
-		if ( ! $email_class->is_enabled() ) {
+		$args['event_gcal_link'] = tribe( Google_Calendar::class )->generate_single_url( $args['event']->ID );
+		$args['event_ical_link'] = tribe_get_single_ical_link( $args['event']->ID );
+
+		if ( ! empty( $args['preview'] ) ) {
+			$args['event_gcal_link'] = '#';
+			$args['event_ical_link'] = '#';
+		}
+
+		tribe( Template::class )->template( 'template-parts/body/event/links', $args, true );
+	}
+
+	/**
+	 * Maybe include event link styles.
+	 *
+	 * @since TBD
+	 *
+	 * @param \Tribe__Template $parent_template Event Tickets template object.
+	 *
+	 * @return void
+	 */
+	public function include_event_link_styles( $parent_template ): void {
+		if ( ! $this->should_show_links( $parent_template, $email_class, $args ) ) {
 			return;
 		}
 
-		$use_ticket_email = tribe_get_option( $email_class->get_option_key( 'use-ticket-email' ), false );
+		tribe( Template::class )->template( 'template-parts/header/head/tec-style', $args, true );
+	}
 
+	/**
+	 * Determines whether or not RSVP should show calendar links.
+	 *
+	 * @since TBD
+	 *
+	 * @param \Tribe__Template $parent_template Event Tickets template object.
+	 *
+	 * @return bool
+	 */
+	public function should_show_links( $parent_template, &$email_class, &$args ): bool {
```

Comment:
Why are we passing the last two parameters as reference?

Reply:
I don't see them being modified in any capacity; there's no need to have them as reference.

Reply:
Ref: https://www.php.net/manual/en/language.references.pass.php

Reply:
`$email_class` and `$args` are null at first, then created within the `should_show_links()` method, so they technically are being modified.

The reason I did it this way is because if the `$email_class` isn't enabled, the the `$args` aren't pulled from the template. I didn't want to create them, if it wasn't needed. Hopefully that makes sense. If there's a better way, let me know.

Reply:
Spoke with @bordoni about this. I was overthinking things. LOL! Thanks!